docs/lifecycle-events.md

```diff
@@ -25,7 +25,9 @@ Fired before the element is removed from the DOM.
 Note that when using this event you are responsible for removing the element yourself.
 
 ```js
-element.parent.removeChild(element)
+if (element.parentNode) {
```

Comment:
I did some more experimenting, and have a better idea of when `onremove` is called.

It seems like you could never have an orphaned child node calling `onremove` to begin with.  `onremove` only gets called when the parent node is staying the same, which makes perfect sense in hindsight.

For example, this will never call the h2's  `onremove` hook, since the `<p>` is changing to a `<div>`.    If both containers were `<div>`, the `onremove` will get called.
```javascript
import {app, h} from 'hyperapp'

function delayedOnRemove(element) {
  console.log('Immediate parent check')
  console.log(element.parentNode)
  setTimeout(() => {
  console.log('Delayed parent check')
    console.log(element.parentNode)
    element.parentNode.removeChild(element)
  }, 3000)
}

app({
  state: 0,
  view: (state, actions) => h('main', null, [
    h('div', {}, [
      h('button', {onclick: () => actions.add() }, 'Add'),
      h('button', {onclick: () => actions.sub() }, 'Sub'),
      h('span', {}, 'Current value is ' + state)
    ]),
    state > 1
    ? h('div', {}, [
      h('h1', { key: '1' }, 'State is greater than 1')
    ])
    : h('p', {}, [
      h('h2', { onremove: delayedOnRemove }, 'State is less than or equal to 1')
    ])
  ]),
  actions: {
    add: state => state + 1,
    sub: state => state - 1
  }
})

```